maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java

```diff
@@ -80,6 +80,12 @@ public FinancialAssessmentEntity update(FinancialAssessmentDTO financialAssessme
             updateChildWeightings(financialAssessment, existingAssessment);
         }
 
+        if (!financialAssessment.getFinAssIncomeEvidences().isEmpty()){
+            financialAssessment.getFinAssIncomeEvidences()
+                    .stream()
+                    .forEach(evidenceDTO -> existingAssessment.addFinAssIncomeEvidences(
+                            assessmentMapper.finAssIncomeEvidenceDTOToFinAssIncomeEvidenceEntity(evidenceDTO)));
+        }
```

Comment:
I think we ought to consider adding some logic to delete income evidence records that are not included in the update request. Right now, if there are two records and the update only contains one, the second records will remain in the DB and be returned by a following find request.

Since this is a `PUT` request, the DB state should reflect the state of the input and not leave old records.

Reply:
Updated

Reply:
It might be easier here to do what we did for Hardships and just use the .clear() method.

```
List<HardshipReviewDetail> detailItems = hardshipReviewDTO.getReviewDetails();
        existing.getReviewDetails().clear();
        if (!detailItems.isEmpty()) {
            detailItems.forEach(
                    detail -> {
                        HardshipReviewDetailEntity reviewDetailEntity =
                                hardshipReviewMapper.hardshipReviewDetailToHardshipReviewDetailEntity(detail);
                        existing.addReviewDetail(reviewDetailEntity);
                    }
            );
        }
```